### PR TITLE
bpo-35843: Implement __getitem__ for _NamespacePath

### PR DESCRIPTION
**Pull request by asottile**

### CODE REVIEW

**Review by eamanu — APPROVED** [review]
LGTM. I re run the sample used on bpo https://bugs.python.org/issue35843 and have the expected behavior.

File: Lib/importlib/_bootstrap_external.py [hunk 1]
```diff
@@ -1163,6 +1163,9 @@ def _recalculate(self):
     def __iter__(self):
         return iter(self._recalculate())
 
+    def __getitem__(self, index):
+        return self._path[index]
```
Comment (serhiy-storchaka) [inline Lib/importlib/_bootstrap_external.py]: Should not we call `self._recalculate()`?

Comment (asottile) [reply Lib/importlib/_bootstrap_external.py]: hmm yeah probably -- let me fix that

**Comment by asottile** [comment]
@serhiy-storchaka should be good to review again -- thanks for the tip :+1:

**Comment by serhiy-storchaka** [comment]
I can say nothing about whether this idea is good or bad. But technically this PR looks correct.

**Review by brettcannon — APPROVED** [review]

**Comment by brettcannon** [comment]
I'm blocking this to see if any other core devs have an opinion on this on the issue tracker, otherwise it LGTM!

**Comment by asottile** [comment]
@brettcannon given the +1 in the bpo issue I think this is good to go

**Comment by brettcannon** [comment]
@asottile that's the plan 😄 Just need to find the time to handle the merge (probably tomorrow/Friday).

**Comment by brettcannon** [comment]
Thanks, @asottile !

**Pull request merged**